Add modules/ and projects/ with monitoring checks

jvperrin opened the pull request:
Here's the monitoring checks and stuff I've set up so far. I'd like to still set up some Jenkins stuff to automatically run a plan on all reviews and run an apply on the main branch, but I haven't done that yet.

I'd also like to make setting up checks less verbose as they're pretty lengthy currently and a lot of the time only a few parameters are actually changing.

jvperrin committed:
Add modules/ and projects/ with monitoring checks

jvperrin committed:
Refactor checks to make them less repetitive to write

jvperrin committed:
Add some more common monitoring checks

jvperrin committed:
Fix some formatting

ja5087 commented:
Looks like tf will support syntax like `for_each` for modules in 0.13: https://github.com/hashicorp/terraform/issues/25016

jvperrin committed:
Make static uptime alert less flaky (trigger count from 1->3)

jvperrin commented:
Ooh nice, that'll make this even more compact and separate out the data a bit more :tada:

jvperrin committed:
Make Jenkins monitoring check more lenient

jvperrin committed:
Tweak auth/keycloak check to be a bit less sensitive

jvperrin committed:
Set all monitoring checks to only trigger after 2 failures

jvperrin committed:
Upgrade to terraform 0.14

jvperrin committed:
Fix alerts not firing, use failing_duration instead

Before, setting trigger_count to above 1 meant that multiple
_timeseries_ were needed for the alert to fire, but that would never
happen, leading the alerts to never appear. Instead, let's use the
duration to determine if the check should fire or not and set it to 2
minutes by default to remove some false positives.

jvperrin committed:
Move terraform binary setup before pre-commit tests

jvperrin committed:
Add plan comments on pull requests

jvperrin commented:
Well not very clean output, but it's something. I'll work some more on that

jvperrin committed:
Make PR comment output less verbose

jvperrin committed:
Upgrade to TF 0.14.6

jvperrin committed:
Run /home/j/jv/jvperrin/repos/terraform/venv/bin/pre-commit install -f --install-hooks
pre-commit installed at .git/hooks/pre-commit
/home/j/jv/jvperrin/repos/terraform/venv/bin/pre-commit run --all-files
Check for added large files..............................................[42mPassed[m
Check for byte-order marker..............................................[42mPassed[m
Check for case conflicts.................................................[42mPassed[m
Check that executables have shebangs.....................................[42mPassed[m
Check JSON...........................................(no files to check)[46;30mSkipped[m
Check for merge conflicts................................................[42mPassed[m
Check Yaml...............................................................[42mPassed[m
Detect Private Key.......................................................[42mPassed[m
Fix End of Files.........................................................[42mPassed[m
Mixed line ending........................................................[42mPassed[m
Trim Trailing Whitespace.................................................[42mPassed[m
Shellcheck Bash Linter...................................................[42mPassed[m
Terraform format.........................................................[42mPassed[m
Terraform validate.......................................................[42mPassed[m, add RO creds for terraform-validate

jvperrin commented:
Heh, now it just needs some markdown code formatting in there, but getting much closer

jvperrin committed:
Add nicer-formatted auto-plan output for GitHub comments

jvperrin committed:
Move output to collapsible sections

nikhiljha commented:
I didn't even know the OCF had terraform configuration...

E: It doesn't; this is new. Phew.

jvperrin commented:
> I didn't even know the OCF had terraform configuration...
> 
> E: It doesn't; this is new. Phew.

Yeah, all new. I essentially didn't want to set up #alerts by doing manual stuff through the console since that can't be searched or changed by people who don't have access. I also figured this could be used for similar stuff to configure the GitHub org (haven't done any of that yet) so that people don't need admin access and that repos can have more standard access controls and settings across the org.

nikhiljha commented on projects/google-cloud/compute-engine.tf:
```diff
@@ -0,0 +1,29 @@
+resource "google_compute_instance" "ocf-offsite" {
```
We're running a VM offsite? What is this for?

nikhiljha commented on projects/google-cloud/compute-engine.tf:
```diff
@@ -0,0 +1,29 @@
+resource "google_compute_instance" "ocf-offsite" {
```
We're running a VM offsite? What is this for?

jvperrin replied:
It was going to be for external monitoring, but we never got around to using it and the actual monitoring checks are probably better than setting up a VM to do it. Anyway, it's in the free tier and has just been running for the last few years without really doing anything, so if you have good ideas for things to do with a small off-site host then this might work well.

nikhiljha commented:
Thanks for working on this! It looks like it'll be really helpful for managing external infrastructure going forward.
